src/Unosquare.Labs.EmbedIO/System.Net/HttpListenerRequestUriBuilder.cs

```diff
@@ -70,47 +64,114 @@ static HttpListenerRequestUriBuilder()
             _cookedUriQuery = cookedUriQuery ?? string.Empty;
         }
 
-        public static Uri GetRequestUri(
-            string rawUri, 
-            string cookedUriScheme, 
-            string cookedUriHost,
-            string cookedUriPath, 
-            string cookedUriQuery)
+        public static Uri GetRequestUri(string rawUri, Uri infoUri)
+        {
+            return new HttpListenerRequestUriBuilder(
+                    rawUri,
+                    infoUri.Scheme,
+                    infoUri.Authority,
+                    infoUri.LocalPath,
+                    infoUri.Query)
+                .Build();
+        }
+
+        private static void AppendOctetsPercentEncoded(StringBuilder target, IEnumerable<byte> octets)
         {
-            var builder = new HttpListenerRequestUriBuilder(
-                rawUri,
-                cookedUriScheme, 
-                cookedUriHost, 
-                cookedUriPath, 
-                cookedUriQuery);
-
-            return builder.Build();
+            foreach (var octet in octets)
+            {
+                target
+                    .Append('%')
+                    .Append(octet.ToString("X2", CultureInfo.InvariantCulture));
+            }
+        }
+
+        private static string GetPath(string uriString)
+        {
+            var pathStartIndex = 0;
+
+            // Perf. improvement: nearly all strings are relative Uris. So just look if the
+            // string starts with '/'. If so, we have a relative Uri and the path starts at position 0.
+            // (http.sys already trimmed leading whitespaces)
+            if (uriString[0] != '/')
+            {
+                // We can't check against cookedUriScheme, since http.sys allows for request http://myserver/ to
+                // use a request line 'GET https://myserver/' (note http vs. https). Therefore check if the
+                // Uri starts with either http:// or https://.
+                var authorityStartIndex = 0;
+                if (uriString.StartsWith("http://", StringComparison.OrdinalIgnoreCase))
```

Comment:
what if...just search indexof "://" and add 3 to it?

Reply:
or even use regex

Reply:
I don't know... the indexOf just introduce another variable...